nixos/modules/system/boot/systemd.nix

```diff
@@ -250,40 +255,28 @@ let
           environment.PATH = config.path;
         }
         (mkIf (config.preStart != "")
-          { serviceConfig.ExecStartPre = makeJobScript "${name}-pre-start" ''
-              #! ${pkgs.runtimeShell} -e
-              ${config.preStart}
-            '';
+          { serviceConfig.ExecStartPre =
+              makeJobScript "${name}-pre-start" config.preStart;
           })
         (mkIf (config.script != "")
-          { serviceConfig.ExecStart = makeJobScript "${name}-start" ''
-              #! ${pkgs.runtimeShell} -e
-              ${config.script}
-            '' + " " + config.scriptArgs;
+          { serviceConfig.ExecStart =
+              makeJobScript "${name}-start" (config.script + " " + config.scriptArgs);
```

Comment:
It's confusing but I think it should be `(makeJobScript "${name}-start" config.script) + " " + config.scriptArgs`

Reply:
Shit, you are right. I thought it was weird when I was writing it...